src/value/ValueTransfer.java

```
@@ -0,0 +1,235 @@
+                case SIGNED_SHIFT_RIGHT:
+                    resultRange = leftRange.signedShiftRight(rightRange);
+                    break;
+                case UNSIGNED_SHIFT_RIGHT:
+                    resultRange = leftRange.unsignedShiftRight(rightRange);
+                    break;
+                case BITWISE_AND:
+                    resultRange = leftRange.bitwiseAnd(rightRange);
+                    break;
+                case BITWISE_OR:
+                    resultRange = leftRange.bitwiseOr(rightRange);
+                    break;
+                case BITWISE_XOR:
+                    resultRange = leftRange.bitwiseXor(rightRange);
+                    break;
+                default:
+                    throw new BugInCF("ValueTransfer: unsupported operation: " + op);
+            }
+            // Any integral type with less than 32 bits would be promoted to 32-bit int type during
+            // operations.
+            return leftNode.getType().getKind() == TypeKind.LONG
+                || rightNode.getType().getKind() == TypeKind.LONG
+                ? resultRange
+                : resultRange.intRange();
+        } else {
+            return Range.EVERYTHING;
+        }
+    }
+
+    /** Get possible integer range from annotation. */
+    private Range getIntRange(Node subNode, TransferInput<CFValue, CFStore> p) {
+        AnnotationMirror val = getValueAnnotation(subNode, p);
+        return getIntRangeFromAnnotation(subNode, val);
+    }
+
+    /**
+     * Returns the {@link Range} object corresponding to the annotation {@code val} casted to the
+     * type of {@code node}.
+     *
+     * @param node a node
+     * @param val annotation mirror
+     * @return the {@link Range} object corresponding to the annotation {@code val} casted to the
+     *     type of {@code node}.
+     */
+    private Range getIntRangeFromAnnotation(Node node, AnnotationMirror val) {
+        Range range;
+        if (val == null
```

Can we throw an error if val is null, at least when not in hack mode?